### DISCUSSION TITLE
OpenModelica

### DISCUSSION THREAD

**Discussion by Urbanek18** [OpenModelica/OpenModelica#10224] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/125592909/219440177-9fb22d0d-8f7a-4a11-a215-55a50f377742.png)

Hi, I have a problem with Open Modelica, when I do simulations it throws me an error after some time. Can you help me?

DDASSL had repeated error test failures on the last attempted step.
can't continue. time = 60.855072
>DASKR--  AT T (=R1) AND STEPSIZE H (=R2) THE                                    
      In above,  R1 =  6.0855071679407E+001   R2 =  5.6436166150552E-014
DASKR--  ERROR TEST FAILED REPEATEDLY OR WITH ABS(H)=HMIN

**Comment by casella** [comment, marked as answer]
The DASSL solver has adaptive error control: it estimates the error when computing a new step, and reduces the step size if that is above the tolerance threshold. In this case DASSL reduced the step size to 5e-14 at time 60.855 and then eventually gave up. This can happen for a number of reasons, typically some of the equations become singular for some reason. I suggest you to simulate until, say, 60.854 or 60.85 and check all the derivatives (type `der` in the simulation result variable filter). If you see some derivative doing something strange when time = 60.855 approaches, then you can figure out why it does by looking at the equations and variables it depends upon.
